code/game/objects/items/tools/flame_tools.dm

```diff
@@ -245,9 +245,16 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 	else if(istype(W, /obj/item/weapon/gun))
 		var/obj/item/weapon/gun/G = W
-		if(istype(G.under, /obj/item/attachable/attached_gun/flamer))
+		if(istype(G, /obj/item/weapon/gun/energy/lasgun))
+			var/obj/item/weapon/gun/energy/lasgun/L = G
+			if(L.cell.charge)
+				light("<span class='notice'>[user] deftly lights their [src] with the [L]'s low power setting.</span>")
+			else
+				to_chat(user, "<span class='warning'>You try to light your [src] with the [L] but your power cell has no charge!</span>")
```

Comment:
Add in a return here or it will still light it up.

Reply:
Why would it? No other lighting source will trigger; consider the pilot light being turned off for the flamer.

Reply:
Sorry, it was simply me saying something erraneous, without taking care to notice there's a light proc for the actual littin the cigarette, which is rightfully uncalled here.